pkg/helmdeployer/deployer.go

```diff
@@ -500,23 +509,63 @@ func getReleaseNameVersionAndNamespace(bundleID, resourcesID string) (string, in
 }
 
 func (h *Helm) getRelease(releaseName, namespace string, version int) (*release.Release, error) {
+	key := &release.Release{
+		Namespace: namespace,
+		Name:      releaseName,
+		Version:   version,
+	}
 
-	hist := action.NewHistory(&h.globalCfg)
-
-	releases, err := hist.Run(releaseName)
-	if err == driver.ErrReleaseNotFound {
-		return nil, ErrNoRelease
-	} else if err != nil {
+	object, found, err := h.releaseCache.Get(key)
+	if err != nil {
 		return nil, err
 	}
+	if !found {
+		// cache MISS, query Helm's storage
+		hist := action.NewHistory(&h.globalCfg)
+		releases, err := hist.Run(releaseName)
+		if err == driver.ErrReleaseNotFound {
+			return nil, ErrNoRelease
+		} else if err != nil {
+			return nil, err
+		}
 
-	for _, release := range releases {
-		if release.Name == releaseName && release.Version == version && release.Namespace == namespace {
-			return release, nil
+		// opportunistically cache all releases in the storage with that name
+		for _, release := range releases {
+			err := h.releaseCache.Add(release)
+			if err != nil {
+				return nil, err
+			}
 		}
+
+		// return the right one if found
+		for _, release := range releases {
+			if release.Name == releaseName && release.Version == version && release.Namespace == namespace {
+				return release, nil
+			}
+		}
+
+		// desired release was not found in Helm's storage. Put a "dummy" in the cache to remember it was not found
+		err = h.releaseCache.Add(&release.Release{
```

Comment:
I wouldn't add a "dummy" release if it was not found. What if a release is not found now, but it becomes available one second later? We would get `ErrNoRelease` because is not in the cache until the cache expires in 5 minutes

Reply:
Based in your research do you think it returns `ErrNoRelease` too often that might impact the performance? I think that it will not happen very often, but I might be wrong

Reply:
I admit I do not have exact numbers but I remember seeing a difference in CPU between builds that had and did not have these lines, at least in the test environment.

Reply:
> I admit I do not have exact numbers but I remember seeing a difference in CPU between builds that had and did not have these lines, at least in the test environment.

then it's ok to leave it if it's affecting performance. Worst case scenario cache will be removed in 5 minutes, and it will work after that